src/domain/configure_ftp_cameras.py

```diff
@@ -104,50 +112,64 @@ def accept_and_close(self):
                                      self.ui.label_key_file_path.text(),
                                      self.ui.label_FTPServer_path.text())
         if cameras:
+            # Check for need of updating cameras credentials. If camera ID changes it is seen as new camera.
             i = 1
             ui_camera_id = []
             while i <= self.num_of_cameras:
-                camera_id_ui = f"lineEdit_camera_id_{i}"
-                camera_id_ln = self.findChild(QLineEdit, camera_id_ui)
-                ui_camera_id.append(camera_id_ln.text())
+                cur_ui_id = self.get_camera_id(i)
+                ui_camera_id.append(cur_ui_id)
                 for camera in cameras:
                     if camera.type == Camera.CameraTypes.FTPCamera:
-                        if camera_id_ln.text() == camera.id:
+                        if cur_ui_id == camera.id:
+                            cur_user = self.get_camera_user(i)
+                            cur_pass = self.get_camera_pass(i)
                             credentials = keyring.get_credential(f"WADAS_FTPcamera_{camera.id}", "")
-                            if credentials and (credentials.username != self.ui.lineEdit_username_1 or
-                            credentials.password != self.ui.lineEdit_password_1):
-                                camera_user_ui = f"lineEdit_username_{i}"
-                                camera_user_ln = self.findChild(QLineEdit, camera_user_ui)
-                                camera_pass_ui = f"lineEdit_password_{i}"
-                                camera_pass_ln = self.findChild(QLineEdit, camera_pass_ui)
-                                keyring.set_password(f"WADAS_FTPcamera_{camera_id_ln.text()}",
-                                                     camera_user_ln.text(),
-                                                     camera_pass_ln.text())
+                            if credentials and (credentials.username != cur_user or credentials.password != cur_pass):
+                                keyring.set_password(f"WADAS_FTPcamera_{cur_ui_id}", cur_user, cur_pass)
                                 break
+                # If camera id is not in cameras list, then is new or modified
+                camera = FTPCamera(cur_ui_id, os.path.join(FTPsServer.ftps_server.ftp_dir, cur_ui_id))
+                cameras.append(camera)
+                # Store credentials in keyring
+                keyring.set_password(f"WADAS_FTPcamera_{cur_ui_id}", self.get_camera_user(i),
+                                     self.get_camera_pass(i))
                 i = i + 1
-            orphan_cameras = [camera for camera in cameras if camera not in ui_camera_id]
+
+            # Check for cameras old id (prior to modification) and remove them
+            orphan_cameras = [camera for camera in cameras if camera.id not in ui_camera_id]
```

Comment:
Replace [] with (): it's more efficient (it uses a generator expression, which gives back an iterator instead of a list).